osu.Game/Modes/UI/HitRenderer.cs

```diff
@@ -28,46 +26,63 @@ public abstract class HitRenderer : Container
         /// </summary>
         public abstract Func<Vector2, Vector2> MapPlayfieldToScreenSpace { get; }
 
-        public abstract bool AllObjectsJudged { get; }
+        /// <summary>
+        /// The number of Judgements required to be triggered
+        /// before the game enters post-play routines.
+        /// </summary>
+        protected abstract int JudgementCount { get; }
+
+        /// <summary>
+        /// The beatmap this HitRenderer is initialized with.
+        /// </summary>
+        protected readonly Beatmap Beatmap;
+
+        private int maxJudgements;
+        private int countJudgements;
+
+        protected HitRenderer(Beatmap beatmap)
+        {
+            Beatmap = beatmap;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            maxJudgements = JudgementCount;
+        }
 
         protected void TriggerOnJudgement(JudgementInfo j)
         {
+            countJudgements++;
+
             OnJudgement?.Invoke(j);
-            if (AllObjectsJudged)
+
+            if (countJudgements == maxJudgements)
                 OnAllJudged?.Invoke();
         }
     }
 
     public abstract class HitRenderer<TObject> : HitRenderer
         where TObject : HitObject
     {
-        private List<TObject> objects;
-
-        protected Playfield<TObject> Playfield;
-
         public override Func<Vector2, Vector2> MapPlayfieldToScreenSpace => Playfield.ScaledContent.ToScreenSpace;
-
-        public override bool AllObjectsJudged => Playfield.HitObjects.Children.First()?.Judgement.Result != null; //reverse depth sort means First() instead of Last().
-
         public IEnumerable<DrawableHitObject> DrawableObjects => Playfield.HitObjects.Children;
 
-        public Beatmap Beatmap
-        {
-            set
-            {
-                objects = Convert(value);
-                if (IsLoaded)
-                    loadObjects();
-            }
-        }
+        protected abstract HitObjectConverter<TObject> Converter { get; }
+        protected virtual List<TObject> Convert(Beatmap beatmap) => Converter.Convert(beatmap);

```

Comment:
why does this still take a beatmap parameter? also this can probably be made private or even inlined in its single usage.